### DISCUSSION TITLE
Unable to Get Correct GitHub User Profile URL in Browser Extension

### DISCUSSION THREAD

**Discussion by GVishnudhasan** [open-sauced-craftwork/browser-extension#28] [Q&A] [answered]
I am facing an issue with getting the correct GitHub user profile URL in my browser extension. When I click the button in the extension, I am trying to extract the GitHub username from the URL. 

But instead of getting the GitHub profile URL (e.g. `https://github.com/bdougie`), I am getting the extension URL (e.g. `chrome-extension://nnpclboejedioknhkpmkiniklhkacgbp/index.html`).

Here's the code of how I am trying to extract the GitHub username from the URL:

`const username = window.location.pathname.split('/')[1];`

I would appreciate any help or suggestions on how to get the correct GitHub user profile URL in my browser extension. Thank you in advance for your help.

**Comment by newtonfav** [comment]
I need the content script file you should get the active tab of the browser using the chrome.tabs API, then you can split the username from the url.

**Comment by akash-singh8** [comment, marked as answer]
Yeah, first you need to add the content script which you can simply do by creating a `script.js` file in your root/main folder and then add this code to `manifest.json` : 
```
"content_scripts": [
    {
      "js": ["/script.js"],
      "matches": ["https://github.com/*"]
}]
```

And inside the script, you can use `window.location.href` to get the current url.

**Reply by GVishnudhasan** [reply]
Yes, that is what I actually tried. My `script.js` file is inside the `/src` directory as it is a react project.

And this is how I included the `script.js` in the `manifest.json`:

```
"content_scripts": [
    {
      "matches": ["https://www.github.com/*"],
      "js": ["/src/scripts.js"]
    }]
```

Still, I am getting the extension URL.

**Reply by Rishav1707** [reply]
Use `document.location` instead of `window.location` may be it works

**Reply by nickytonline** [reply]
What @akash-singh8 mentioned. You can access the URL like you normally would, assuming you’re in the content script (not a script related to an extension pop up or a background script).

When you’re in a content script, you can do all the things you can do in a site’s client-side scripts.

**Comment by Aryan-Srivastava** [comment]
Hey @GVishnudhasan , I'm facing the same issue. Did you find any solution for this?

**Comment by Mayank-Tripathi32** [comment]
@Aryan-Srivastava you can get the name from content script, you can also get it by 
```
let user = null;
  if (
    document.getElementsByClassName("vcard-username")[0] !== (null || undefined)
  ) {
    user = document.getElementsByClassName("vcard-username")[0].innerText;
  }
```
You have to do that inside content script or it will not work.
